zerver/tests/test_unread.py

```diff
@@ -137,3 +137,109 @@ def test_update_all_flags(self):
         for msg in self.get_old_messages():
             self.assertEqual(msg['flags'], [])
 
+    def test_mark_all_in_stream_read(self):
+        self.login("hamlet@zulip.com")
+
+        result = self.client.post("/json/messages/flags", {"messages": ujson.dumps([]),
+                                                           "op": "remove",
+                                                           "flag": "read",
+                                                           "stream_name": "Scotland"})
+
+        self.assert_json_success(result)
+        user_profile = get_user_profile_by_email("hamlet@zulip.com")
+        stream = get_stream("Scotland", user_profile.realm)
+        recipient = get_recipient(Recipient.STREAM, stream.id)
+        message_ids = list(UserMessage.objects.filter(message__recipient=recipient,
+                                                    user_profile=user_profile
+        ).values_list('message__id',flat=True))
```

Comment:
See my comment below about structure, but if you're going to write the tests this way, somewhere it should be asserted that this len(message_ids) > 0; otherwise this whole test could be vaccuous if `Scotland` doesn't have any messages on it :)